### PR TITLE
Implement "/product" API endpoint

### PR DESCRIPTION
**Pull request by tahelmo**
In this pull request I implemented the [CRUD](https://github.com/LateBox/server/blob/8f6c0a483a7b1d63fff5f94333c306f46c6f5b36/src/main/java/com/latebox/server/product/ProductController.java) endpoint for the /product object

I also did the database configuration in another [another commit](https://github.com/LateBox/server/blob/8f6c0a483a7b1d63fff5f94333c306f46c6f5b36/src/main/resources/application.properties).

### CODE REVIEW

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    @GetMapping("/products/{id}")
+    Product one(@PathVariable Long id) {
+
+        return repository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(id));
+    }
+
+    @PutMapping("/products/{id}")
+    Product replaceEmployee(@RequestBody Product newEmployee, @PathVariable Long id) {
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:34]:
```suggestion
    Product replaceProduct(@RequestBody Product newProduct, @PathVariable Long id) {
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        return repository.findById(id)
+                .map(employee -> {
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:37]:
```suggestion
                .map(product-> {
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    employee.setName(newEmployee.getName());
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:38]:
```suggestion
                    product.setName(newProduct.getName());
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                    employee.setDescription(newEmployee.getDescription());
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:39]:
```suggestion
                    product.setDescription(newProduct.getDescription());
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                    employee.setPrice(newEmployee.getPrice());
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:40]:
```suggestion
                    product.setPrice(newProduct.getPrice());
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                    employee.setRestaurantId(newEmployee.getRestaurantId());
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:41]:
```suggestion
                    product.setRestaurantId(newProduct.getRestaurantId());
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                    return repository.save(employee);
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:42]:
```suggestion
                    return repository.save(product);
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                })
+                .orElseGet(() -> {
+                    newEmployee.setId(id);
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:45]:
```suggestion
                    newProduct.setId(id);
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+                    return repository.save(newEmployee);
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductController.java:46]:
```suggestion
                    return repository.save(newProduct);
```

File: src/main/java/com/latebox/server/product/ProductNotFoundAdvice.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+class ProductNotFoundAdvice {
+
+    @ResponseBody
+    @ExceptionHandler(ProductNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    String employeeNotFoundHandler(ProductNotFoundException ex) {
```
Comment (uygaraktuna) [inline src/main/java/com/latebox/server/product/ProductNotFoundAdvice.java:15]:
```suggestion
    String productNotFoundHandler(ProductNotFoundException ex) {
```

File: src/main/java/com/latebox/server/product/ProductController.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+                });
+    }
+
+    @DeleteMapping("/products/{id}")
+    void deleteProduct(@PathVariable Long id) {
+        repository.deleteById(id);
```
[the remark this answers is not in the document]
Comment (tahelmo) [reply src/main/java/com/latebox/server/product/ProductController.java]: Yes I believe we need to do it.
Fixed in [d907d18](https://github.com/LateBox/server/pull/9/commits/d907d18b28212c65f91587e49ceae00f0b8eaf13)

**Review by uygaraktuna — APPROVED** [review]

**Pull request merged**